udp/src/main/scala/akka/stream/alpakka/udp/impl/UdpBind.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.udp.impl
+
+import java.net.InetSocketAddress
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.annotation.InternalApi
+import akka.io.{IO, Udp}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.stream.alpakka.udp.UdpMessage
+import akka.stream.stage._
+
+import scala.concurrent.{Future, Promise}
+
+/**
+ * Binds to the given local address using UDP manager actor.
+ */
+@InternalApi private[udp] final class UdpBindLogic(localAddress: InetSocketAddress,
+                                                   boundPromise: Promise[InetSocketAddress])(
+    val shape: FlowShape[UdpMessage, UdpMessage]
+)(implicit val system: ActorSystem)
+    extends GraphStageLogic(shape) {
+
+  private def in = shape.in
+  private def out = shape.out
+
+  private var listener: ActorRef = _
+
+  override def preStart(): Unit = {
+    implicit val sender = getStageActor(processIncoming).ref
+    IO(Udp) ! Udp.Bind(sender, localAddress)
```

Comment:
Would this stage need some binding timeout support?

Reply:
I am not sure. If, for some reason, binding fails, the UDP manager actor will send back `Udp.CommandFailed` message, which this stage is going to handle.